enhancements/improve-kubeletconfig-default-configuration-check-enhancement.md

```diff
@@ -0,0 +1,286 @@
+  ],
+  "tlsMinVersion": "VersionTLS12",
+  "rotateCertificates": true,
+  "serverTLSBootstrap": true,
+  "registryPullQPS": 5,
+  "registryBurst": 10,
+  "eventRecordQPS": 5,
+  "eventBurst": 10,
+  "enableDebuggingHandlers": true,
+  "streamingConnectionIdleTimeout": "4h0m0s",
+  "cpuManagerReconcilePeriod": "10s",
+  "memoryManagerPolicy": "None",
+  "topologyManagerPolicy": "none",
+  "topologyManagerScope": "container",
+  "runtimeRequestTimeout": "2m0s",
+  "hairpinMode": "promiscuous-bridge",
+  "maxPods": 250,
+  "podPidsLimit": 4096,
+  "resolvConf": "/etc/resolv.conf",
+  "cpuCFSQuota": true,
+  "cpuCFSQuotaPeriod": "100ms",
+  "nodeStatusMaxImages": 50,
+  "maxOpenFiles": 1000000,
+  "contentType": "application/vnd.kubernetes.protobuf",
+  "kubeAPIQPS": 50,
+  "kubeAPIBurst": 100,
+  "serializeImagePulls": false,
+  "evictionHard": {
+    "imagefs.available": "15%",
+    "memory.available": "100Mi",
+    "nodefs.available": "10%",
+    "nodefs.inodesFree": "5%"
+  },
+  "evictionPressureTransitionPeriod": "5m0s",
+  "enableControllerAttachDetach": true,
+  "makeIPTablesUtilChains": true,
+  "iptablesMasqueradeBit": 14,
+  "iptablesDropBit": 15,
+  "featureGates": {
+    "APIPriorityAndFairness": true,
+    "CSIMigrationAWS": false,
+    "CSIMigrationAzureFile": false,
+    "CSIMigrationGCE": false,
+    "CSIMigrationvSphere": false,
+    "DownwardAPIHugePages": true,
+    "PodSecurity": true,
+    "RotateKubeletServerCertificate": true
+  },
+  "failSwapOn": true,
+  "memorySwap": {},
+  "containerLogMaxSize": "50Mi",
+  "containerLogMaxFiles": 5,
+  "configMapAndSecretChangeDetectionStrategy": "Watch",
+  "systemReserved": {
+    "cpu": "500m",
+    "memory": "1Gi"
+  },
+  "enforceNodeAllocatable": [
+    "pods"
+  ],
+  "volumePluginDir": "/etc/kubernetes/kubelet-plugins/volume/exec",
+  "logging": {
+    "format": "text",
+    "flushFrequency": 5000000000,
+    "verbosity": 2,
+    "options": {
+      "json": {
+        "infoBufferSize": "0"
+      }
+    }
+  },
+  "enableSystemLogHandler": true,
+  "shutdownGracePeriod": "0s",
+  "shutdownGracePeriodCriticalPods": "0s",
+  "enableProfilingHandler": true,
+  "enableDebugFlagsHandler": true,
+  "seccompDefault": false,
+  "registerNode": true,
+  "kind": "KubeletConfiguration",
+  "apiVersion": "kubelet.config.k8s.io/v1beta1"
+}
+```
+
+When we get `KubeletConfig` for each node, we will compare them, and
+only save the consistent result to a file at `/api/v1/nodes/proxy/configz`
+
+In the CaC content, we just need to check one location `/api/v1/nodes/proxy/configz`
+for `KubeletConfig` default configuration
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters with many nodes, I'd like to run
+  compliance scans on my fleet in such a way that we have less compliance rule failures when
+  no reboot remediation has been applied.
+
+### API Extensions
+
+None is changed
+
+
+### Risks and Mitigations
+
+There are no risks or mitigations for this change.
```

Comment:
We're writing consolidation logic and we have to decide what we do with differences, right?

This might be tough to answer, but how likely is it that configs will be different? If they are different, how are we going to decide which configuration to keep, if any?

Reply:
there can be an easy way if there is a difference, we can just leave that specific configuration out, only save the consistent result and we will instruct the user to find out which node is in question by querying the KubeletConfig for each node